mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java

```diff
@@ -2509,5 +2526,98 @@ public void updateForNewTable(ContentValues bookmark) {
+            contentValues.put(Bookmarks.SYNC_VERSION, 0);
+            contentValues.put(Bookmarks.LOCAL_VERSION, 1);
+            db.insert(Bookmarks.TABLE_NAME, null, contentValues);
+        }
+        rowCursor.close();
+    }
+
+    private void migrateHistory(SQLiteDatabase db) {
+        final HashMap<GhosteryHistoryItem, ArrayList<Long> > historyItems = new HashMap<>();
+        final Cursor c = db.rawQuery("SELECT * FROM ghostery.history", null);
+        while(c.moveToNext()) {
+            final String url = c.getString(c.getColumnIndex("url"));
+            final String title = c.getString(c.getColumnIndex("title"));
+            final long date = c.getLong(c.getColumnIndex("date"));
+            final GhosteryHistoryItem ghosteryHistoryItem = new GhosteryHistoryItem(url, title);
+            if(historyItems.containsKey(ghosteryHistoryItem)) {
+                final ArrayList<Long> dateList = historyItems.get(ghosteryHistoryItem);
+                dateList.add(date);
+            } else {
+                final ArrayList<Long> dateList = new ArrayList<>();
+                dateList.add(date);
+                historyItems.put(ghosteryHistoryItem, dateList);
+            }
+        }
+        c.close();
+        for (Map.Entry<GhosteryHistoryItem, ArrayList<Long> > entry : historyItems.entrySet()) {
+            final GhosteryHistoryItem ghosteryHistoryItem = entry.getKey();
+            final ArrayList<Long> timeStamps = entry.getValue();
+            Collections.sort(timeStamps);
+            final String guid = Utils.generateGuid();
+            final long creationDate = timeStamps.get(0);
+            final long modifiedDate = timeStamps.get(timeStamps.size()-1);
+            final ContentValues historyValues = new ContentValues();
+            historyValues.put(History.TITLE, ghosteryHistoryItem.title);
+            historyValues.put(History.URL, ghosteryHistoryItem.url);
+            historyValues.put(History.VISITS, timeStamps.size());
+            historyValues.put(History.LOCAL_VISITS, timeStamps.size());
+            historyValues.put(History.REMOTE_VISITS, 0);
+            historyValues.put(History.DATE_LAST_VISITED, modifiedDate);
+            historyValues.put(History.LOCAL_DATE_LAST_VISITED, modifiedDate);
+            historyValues.put(History.REMOTE_DATE_LAST_VISITED, 0);
+            historyValues.put(History.DATE_CREATED, creationDate);
+            historyValues.put(History.DATE_MODIFIED, modifiedDate);
+            historyValues.put(History.GUID, guid);
+            historyValues.put(History.IS_DELETED, 0);
+            db.insert(History.TABLE_NAME, null, historyValues);
+            for(long timestamp : timeStamps) {
+                final ContentValues visitsValues = new ContentValues();
+                visitsValues.put(Visits.HISTORY_GUID, guid);
+                visitsValues.put(Visits.VISIT_TYPE, 1);
+                visitsValues.put(Visits.DATE_VISITED, timestamp*1000);
```

Comment:
I suppose this value is a `long`, replace with `timestamp*1000L`

Reply:
long multiplied by int will result in a long :)